connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/querytest/GeneratedSubquerySuite.scala

```diff
@@ -182,8 +182,11 @@ class GeneratedSubquerySuite extends DockerJDBCIntegrationSuite with QueryGenera
     } else {
```

Comment:
I should add some other predicates like `NotEquals`, `LessThan`... (reuse all newly added ones) here, is this right? @andylam-db

Reply:
Not sure I understand what you mean by `where the upper layer calls`.

Reply:
Oh, sorry, I didn't make it clear, I meant in line359-362, 
`for {
        subqueryOperator <- subqueryOperators
        isDistinct <- distinctChoices(subqueryOperator)
      }`

add another axis about `correlationConditions` here to cover all the different predicates, not only hardcoded about `=` in `generateQuery` method.

Reply:
Sounds great!